core/domain/user_services.py

```diff
@@ -645,9 +741,11 @@ def get_user_settings_by_gae_id(gae_id, strict=False):
     Raises:
         Exception. The value of strict is True and given gae_id does not exist.
     """
-    user_settings_model = user_models.UserSettingsModel.get_by_gae_id(gae_id)
-    if user_settings_model is not None:
-        user_settings = _transform_user_settings(user_settings_model)
+    user_auth_details_model = user_models.UserAuthDetailsModel.get_by_auth_id(
+        feconf.AUTH_METHOD_GAE, gae_id)
+    if user_auth_details_model is not None:
+        user_settings = _transform_user_settings(
```

Comment:
Can we call this something like _get_user_settings_from_model()? I have no idea what the current name represents, and had to look it up.

Reply:
Sure. I have now changed it to what you suggested.